hivemq-edge/src/main/java/com/hivemq/bridge/mqtt/RemoteMqttForwarder.java

```diff
@@ -216,14 +236,17 @@ private synchronized void sendPublishToRemote(
         final Mqtt5Publish mqtt5Publish = convertPublishForClient(publish);
         final CompletableFuture<Mqtt5PublishResult> publishResult =
                 remoteMqttClient.getMqtt5Client().publish(mqtt5Publish);
+        final OutflightPublishInformation outflightPublishInformation =
+                new OutflightPublishInformation(queueId, publish.getUniqueId());
+        outflightQueue.add(outflightPublishInformation);
         publishResult.whenComplete((mqtt5PublishResult, throwable) -> {
             if (throwable != null) {
                 handlePublishError(publish, throwable);
-                finishProcessing(originalQoS, publish.getUniqueId(), queueId);
             } else {
                 perBridgeMetrics.getPublishForwardSuccessCounter().inc();
-                finishProcessing(originalQoS, publish.getUniqueId(), queueId);
             }
+            outflightQueue.remove(outflightPublishInformation);
```

Comment:
minor: would be good if we keep the same order between outflightQueue.remove and finishProcessing here and in line 260+